plugins/mac-virtualcam/src/obs-plugin/plugin-main.mm

```diff
@@ -9,33 +12,164 @@
 	return "macOS virtual webcam output";
 }
 
+NSString *const OBSDalDestination = @"/Library/CoreMediaIO/Plug-Ins/DAL";
+
+static bool cmio_extension_supported()
+{
+	if (@available(macOS 13.0, *)) {
+		return true;
+	} else {
+		return false;
+	}
+}
```

Comment:
I wonder if this also needs some kind of `#if CAMERA_EXTENSION_AVAILABLE` define in addition to the runtime check, because as I understand (and correct me if I'm wrong) one is still able to compile OBS themselves without the camera extension(?). In such a case it would try to install/use an extension that doesn't exists.

Reply:
In that case one has to configure the project with `ENABLE_VIRTUALCAM` set to `FALSE` to disable that functionality entirely.

Anyone with a valid Apple Developer certificate (or being a member of an Apple Developer team) can compile obs-studio with the extension and also run the build locally (no SIP disabling necessary). The only limitation is generating a signed build for distribution (the "Manual" code singing style), which requires changing the bundle identifier to a different value for which one has a valid provisioning profile.